src/Microsoft.Graph.Cli.Core/Http/UriReplacementHandler.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph.Cli.Core.Http;
+
+/// <summary>
+/// Interface for making URI replacements.
+/// </summary>
+public interface IUriReplacement
+{
+    /// <summary>
+    /// Accepts a URI and returns a new URI with all replacements applied.
+    /// </summary>
+    /// <param name="original">The URI to apply replacements to</param>
+    /// <returns>A new URI with all replacements applied.</returns>
+    Uri? Replace(Uri? original);
+}
+
+/// <summary>
+/// Specialized replacement for /[version]/users/me with /[version]/me
+/// </summary>
+public struct MeUriReplacement : IUriReplacement
+{
+    /// <summary>
+    /// If a URI path starts with /[version]/users/me, replace it with /[version]/me
+    /// </summary>
+    /// <param name="original">The original URI</param>
+    /// <returns>A URI with /[version]/users/me replaced with /[version]/me</returns>
+    /// <remarks>This method assumes that the first segment after the root is a version segment to match Microsoft Graph API's format.</remarks>
+    public readonly Uri? Replace(Uri? original)
+    {
+        if (original is null)
+        {
+            return null;
+        }
+
+        if (original.Segments.Length < 4)
+        {
+            // Must have at least segments "/", "[version]/", "users/", "me"
+            return original;
+        }
+
+        Span<char> toMatch = stackalloc char[9];
```

Comment:
why declare/alloc these constants on every execution? couldn't it be a set of static private fields instead?

Reply:
I don't think stack allocation can happen at the object level. This is the same as having `string toMatch = "/users/me"` but on the stack.

Reply:
We'd have to fall back to multiple heap allocations which achieve the same goal. Then we'd just as well change back to using strings over spans.

Reply:
And doing multiple stackalloc is not a concern since the process is short lived. Got it, thanks.